scripts/build_egs_potentials.py

```diff
@@ -0,0 +1,181 @@
+        .set_index(['x', 'y'])
+        .rename(columns={'value': 'capex'})
+    )
+    gen_gdf = (
+        get_raster_file(snakemake.input.egs_gen)
+        .set_index(['x', 'y'])
+        .rename(columns={'value': 'gen'})
+    )
+    opex_gdf = (
+        get_raster_file(snakemake.input.egs_opex)
+        .set_index(['x', 'y'])
+        .rename(columns={'value': 'opex'})
+    )
+
+    gdf = capex_gdf[['capex', 'geometry']].join(gen_gdf[['gen']], how='inner')
+    gdf = gdf.join(opex_gdf[['opex']], how='inner')
+
+    gdf['plant_capacity'] = gdf['gen'].div(8760)
+    gdf['capex($/kWel)'] = gdf['capex'].div(gdf['plant_capacity']).mul(1e3)
+    gdf['plant_capacity(MWel)'] = gdf['plant_capacity']
+    gdf['opex[$/kWh]'] = gdf['opex'].mul(1e-2)
+
+    gdf.rename(
+        columns={
+            'capex($/kWel)': 'capex[$/kW]',
+            'plant_capacity(MWel)': 'available_capacity[MW]'
+            }, inplace=True)
+    
+    gdf = gdf[['capex[$/kW]', 'opex[$/kWh]', 'available_capacity[MW]', 'geometry']]
+
+    nodal_egs_potentials = pd.DataFrame(
+        np.nan,
+        index=regions.index,
+        columns=['capex[$/kW]', 'opex[$/kWh]', 'p_nom_max[MW]']
+        )
+
+    config = snakemake.params["enhanced_geothermal"]
+
+    regional_potentials = []
+
+    for name, geom in tqdm(
+        regions.geometry.items(),
+        desc='Matching EGS potentials to network regions',
+        ascii=True
+        ):
+
+        ss = gdf.loc[gdf.geometry.within(geom)]
+        if ss.empty:
+            continue
+
+        ss = (
+            ss[['capex[$/kW]', 'opex[$/kWh]', 'available_capacity[MW]']]
+            .reset_index(drop=True)
+            .sort_values(by='capex[$/kW]')
+        )
+
+        ss['agg_available_capacity[MW]'] = ss['available_capacity[MW]'].cumsum()
+
+        bins = pd.Series(
+                np.linspace(
+                    ss['capex[$/kW]'].min(),
+                    ss['capex[$/kW]'].max(),
+                    config['max_levels']+1
+                    )
+                )
+
+        labels = bins.rolling(2).mean().dropna().tolist()
+
+        if ss.empty:
+            continue
+
+        if len(ss) > 1:
+            ss['level'] = pd.cut(
+                ss['capex[$/kW]'],
+                bins=bins,
+                labels=labels,
+                duplicates='drop'
+                )
+        else:
+            ss['level'] = ss['capex[$/kW]'].iloc[0]
+
+        ss = ss.dropna()
+
+        ss = (
+            ss
+            .groupby('level', observed=False)
+            [['available_capacity[MW]', 'opex[$/kWh]']]
+            .agg({'available_capacity[MW]': 'sum', 'opex[$/kWh]': 'mean'})
+        )
+        ss.index = list(map(myround, ss.index))
+        ss.index = (
+            pd.MultiIndex.from_product(
+                [[name], ss.index],
+                names=['network_region', 'capex[$/kW]']
+                )
+            )
+
+        regional_potentials.append(ss) 
```

Comment:
if in this analysis we create a new column "capital_cost" accounting for the parameter of the discount rate from the config file, we can move all the calculations here and then use in add_electricity the column "capital_cost" directly.